paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerAdapter.kt

```diff
@@ -179,35 +179,37 @@ interface CustomerAdapter {
 
     @ExperimentalCustomerSheetApi
     @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-    @JvmInline
-    value class Result<out T> internal constructor(
-        internal val value: Any?
-    ) {
+    sealed class Result<T> {
+
+        @ExperimentalCustomerSheetApi
+        @get:RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        val isFailure: Boolean get() = this is Failure
 
-        val isSuccess: Boolean get() = value !is Failure
-        val isFailure: Boolean get() = value is Failure
+        @ExperimentalCustomerSheetApi
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        data class Success<T>(
```

Comment:
If we end up exposing this, we should use a `class` and restrict the constructor.